File: Programming Fundamentals with Python/Functions/password_validator.py

def is_long(pass_input):
    if 6 <= len(pass_input) <= 10:
        return True
    else:
        print("Password must be between 6 and 10 characters")

def is_letters_digits(pass_input):
    for char in pass_input:
        if not char.isalnum():
            print("Password must consist only of letters and digits")
            return False
    return True

def have_digits(pass_input):
    digit_count = sum(1 for char in pass_input if char.isdigit())
    if digit_count >= 2:
        return True
    else:
        print("Password must have at least 2 digits")
        return False

def validate_password(pass_input):
    if is_long(pass_input) and is_letters_digits(pass_input) and have_digits(pass_input):
        print("Password is valid")


pass_input = input()
validate_password(pass_input)

File: Programming Fundamentals with Java/Methods/PasswordValidator.java
import java.util.Scanner;

public class PasswordValidator 
{
    private static boolean isValidPassword(String password) 
    {
        // 6-10 characters
        String lengthRegex = "^.{6,10}$";

        // Consists only of letters and digits
        String characterRegex = "^[a-zA-Z0-9]+$";

        // Have at least 2 digits
        String digitRegex = ".*\\d.*\\d.*";

        return password.matches(lengthRegex) && password.matches(characterRegex) 
                && password.matches(digitRegex);
    }

    private static void printInvalidPasswordMessages(String password) 
    {
        // Rule 1: 6-10 characters
        if (!password.matches("^.{6,10}$")) 
        {
            System.out.println("Password must be between 6 and 10 characters");
        }

        // Rule 2: Consists only of letters and digits
        if (!password.matches("^[a-zA-Z0-9]+$")) 
        {
            System.out.println("Password must consist only of letters and digits");
        }

        // Rule 3: Have at least 2 digits
        if (!password.matches(".*\\d.*\\d.*")) 
        {
            System.out.println("Password must have at least 2 digits");
        }
    }

    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);

        String password = input.nextLine();
        if (isValidPassword(password)) 
        {
            System.out.println("Password is valid");
        } 
        else 
        {
            printInvalidPasswordMessages(password);
        }
    }
}
